src/components/markdown/editor.tsx

```diff
@@ -0,0 +1,121 @@
+import "@mdxeditor/editor/style.css";
+
+import {
+  AdmonitionDirectiveDescriptor,
+  BlockTypeSelect,
+  BoldItalicUnderlineToggles,
+  codeBlockPlugin,
+  codeMirrorPlugin,
+  CodeToggle,
+  CreateLink,
+  diffSourcePlugin,
+  DiffSourceToggleWrapper,
+  directivesPlugin,
+  imagePlugin,
+  InsertAdmonition,
+  InsertCodeBlock,
+  InsertImage,
+  InsertTable,
+  InsertThematicBreak,
+  linkDialogPlugin,
+  linkPlugin,
+  listsPlugin,
+  ListsToggle,
+  markdownShortcutPlugin,
+  MDXEditor,
+  quotePlugin,
+  Separator,
+  StrikeThroughSupSubToggles,
+  tablePlugin,
+  thematicBreakPlugin,
+  toolbarPlugin,
+  UndoRedo,
+} from "@mdxeditor/editor";
+import { headingsPlugin } from "@mdxeditor/editor";
+
+type MarkdownEditorProps = {
+  markdown: string;
+  diffMarkdown?: string | undefined;
+  onChange?: ((markdown: string, initialMarkdownNormalize: boolean) => void) | undefined;
+};
+
+const codeBlockLanguages = {
+  py: "Python",
+  cpp: "C++",
+  java: "Java",
+  rb: "Ruby",
+  go: "Go",
+  rust: "Rust",
+  js: "JavaScript",
+  css: "CSS",
+  html: "HTML",
+};
+
+export const MarkdownEditor: React.FC<MarkdownEditorProps> = ({ markdown, diffMarkdown, onChange }) => {
+  const onChangeInternal = (markdown: string, initialMarkdownNormalize: boolean): void => {
+    // Post-process the markdown to unescape _ in math Latex block
+    const mathPattern: RegExp = /\$\$?\n*.*?\n*\$?\$/gm;
+    const matches = markdown.matchAll(mathPattern);
+
+    for (const match of matches) {
+      // We use split-join because replaceAll is only supported on es2021 or later
+      // In addition, replace also somehow replaces $$ with $
+      const unescaped = match[0].split("\\_").join("_");
+      markdown = markdown.split(match[0]).join(unescaped);
+    }
+    if (onChange) onChange(markdown, initialMarkdownNormalize);
+  };
+
+  return (
+    <div className="rounded-sm bg-white">
+      <MDXEditor
+        markdown={markdown}
```

Comment:
```suggestion
      <MDXEditor
        className="rounded-sm bg-white"
        markdown={markdown}
```

Reply:
Done. 

I was planning to do this but somehow Zed said that className was not available before. Weird.